java/src/jmri/jmrit/operations/trains/TrainSwitchListEditFrame.java

```diff
@@ -257,8 +257,9 @@ public void checkBoxActionPerformed(java.awt.event.ActionEvent ae) {
 
     // Remove all terminated or reset trains from the switch lists for selected locations
     private void reset() {
-        for (JCheckBox checkbox : locationCheckBoxes) {
-            String locationName = checkbox.getName();
+        // this for loop prevents ConcurrentModificationException when printing and status changes
+        for (int i = 0; i < locationCheckBoxes.size(); i++) {
```

Comment:
You can also fix this using ```for (JCheckBox checkbox : new ArrayList(locationCheckBoxes)) { ... }```; although iterating over the list using a counter as you do avoids the exception, it does not prevent the locationCheckBoxes from changing under you as you iterate over it, the above form ensures that every item in the list is iterated over.